### PR TITLE
Parquet-2030: Expose page size row check configurations to ParquetWri…

### PR DESCRIPTION
**Pull request by Miksu82**
This MR fixes https://issues.apache.org/jira/browse/PARQUET-2030. It exposes `page.size.row.check.max` and `page.size.row.check.max` to `ParquetWriter.Builder`

### CODE REVIEW

**Review by gszadovszky — CHANGES_REQUESTED** [review]
Thanks for working on this. I have some notes about the tests. 
(Note: we do not need to validate the whole feature of the page size check only the fact that the properties have passed properly.)

**Comment by Miksu82** [comment]
> We do not need to validate the whole feature of the page size check only the fact that the properties have passed properly

Yeah you are right, that should be in `InternalParquetRecordWriter` unit tests. I'll try to simplify the tests I've added so that they only test that the property has been set.

File: parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetWriter.java [hunk 1]
```diff
@@ -279,4 +285,133 @@ public void testParquetFileWithBloomFilter() throws IOException {
...
+    writer.write(factory.newGroup().append("str", "bar"));
+    writer.write(factory.newGroup().append("str", "baz"));
+  }
+
+  private void assertHasNotFlushed(TestOutputFile file) {
+    int emptyFileLength = ParquetFileWriter.MAGIC.length;
+    assertEquals(emptyFileLength, file.stats.getBytesWritten());
+  }
+
+  private void assertHasFlushed(TestOutputFile file, int numWrites, int numFlushes) {
```
[the remark this answers is not in the document]
Comment (Miksu82) [reply parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetWriter.java:411]: Hmm..no. Everytime a record is flushed it writes the actual record + the metadata to indicate that a row group has ended (the writer is created by defining `rowGroupSize=1`).

It works like this

```
write("foo") // writes 7B
flush() // writes 23B
write("bar") // writes 7B
flush() // writes 23B

// Totally has written:  60B
```

but

```
write("foo") // writes 7B
write("bar") // writes 7B
flush()  // writes 23B

// Totally has written: 37
```

And by defining the `page.size.row.check.min` and `page.size.row.check.max` I can control how often the `InternalParquetRecordWriter` flushes the records, and I can assert that by calculating the number of bytes written in total.

But you are absolutely right, that this is complicated and in the wrong place so I'll remove these test and just assert that the properties have been set.

**Comment by Miksu82** [comment]
Unfortunately I cannot find a way to only assert that properties have been set. It probably could be done with extensive refactoring but I don't see the point to do that in this test. Also other property setters in the builder are not unit tested so I guess I could just remove these tests completely. 

However I did notice that I can assert the number of flushes by counting the number of blocks in the Parquet file footer. Asserting that way would make the test easier to understand.

Let me know what you prefer. Remove the tests or fix the assertations? I'll continue after that.

File: parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetWriter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Miksu82) [reply parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetWriter.java:411]: Seems really hard to assert that the properties are just set. See my comment below

**Comment by gszadovszky** [comment]
Yeah, checking only that properties are really passed is not easy. I did not want to say to remove all the tests completely (or any) just that we don't need to test every aspect of the original feature. I think, it is enough to test the difference between setting the properties and not so you know they are passed. You can see that passing such properties are not that obvious so I would keep some test to be sure.

The idea of checking the number of row groups is a good point.

**Comment by Miksu82** [comment]
I now have two tests. One with default params and one with changes params. And I assert by checking the number of blocks in the footer.

I also removed the changes I did for `TestOutputFile` since they are not needed anymore.

**Review by gszadovszky — APPROVED** [review]
Thanks a lot for your efforts!

**Pull request merged**